app/controllers/homes_controller.rb

```diff
@@ -1,7 +1,28 @@
 class HomesController < ApplicationController
   def index
-    @pagy, @products = pagy Product.newest, items: Settings.items_per_page
+    @products = Product.newest
+    @products = @products.search_by_name params[:name] if params[:name].present?
```

Comment:
```suggestion
    @products = Product.search_by_name(params[:name]).newest
```
ri hì, chỗ check param name nên để trong scope làm
`scope :search_by_name,`
 `       ->(name){where("LOWER(name) LIKE ?", "%#{name.downcase}%" if name.present?)}`
`